src/board.py

```diff
@@ -144,31 +148,27 @@ def square_of(self, piece: Piece | None) -> Square | None:
                     return Square(Vector(rank, file))  # HACK: I do not like that I have to chain constructors like this.
 
     
-    def _make_condition(self, piece: Piece, captured: bool) -> Callable[[Square], bool]:
+    def _make_condition(self, piece: Piece) -> Callable[[Square], tuple[bool, Piece]]:
         """Create a condition function
         """
-        def condition(target_square: Square) -> bool:
+        def condition(target_square: Square) -> tuple[bool, Piece]:
```

Comment:
I return a piece in order to take into account the case where a `Piece` is captured. Otherwise, `Range` pieces will continue to be able to move over them. This was allowed by `legal_moves` before.